src/Raven.Server/Documents/Sharding/Handlers/Admin/ShardedAdminShardingHandler.cs

```diff
@@ -13,5 +29,182 @@ public async Task ExecuteMoveDocuments()
                                                                              "This operation is available only from a specific shard"))
                 await processor.ExecuteAsync();
         }
+
+        [RavenShardedAction("/databases/*/admin/sharding/prefixed/add", "POST")]
+        public async Task AddPrefixedShardingSetting()
+        {
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            {
+                var json = await context.ReadForMemoryAsync(RequestBodyStream(), GetType().Name);
+                var setting = JsonDeserializationCluster.PrefixedShardingSetting(json);
+                var shardingConfiguration = ServerStore.Cluster.ReadShardingConfiguration(DatabaseName);
+
+                ShardingStore.AssertValidPrefix(setting, shardingConfiguration);
+
+                var exists = shardingConfiguration.Prefixed.BinarySearch(setting, PrefixedSettingComparer.Instance) >= 0;
+                if (exists)
+                    throw new InvalidOperationException(
+                        $"Prefix '{setting.Prefix}' already exists in {nameof(ShardingConfiguration)}.{nameof(ShardingConfiguration.Prefixed)}. please use '{nameof(UpdatePrefixedShardingSettingOperation)} operation'");
```

Comment:
please => Please
add `.` (dot`) at the end.
there is a `'` there at the end as well?